FIX: validation alert not shown when user enters spaces as input in name and password fields(#202)

sachinsom93 opened the pull request:
### Description
Validated name, password, confirm password field in register page for spaces as input using regex patterns.

Fixes #202 

### Type of Change:
<!--**Delete irrelevant options.**-->

- Code
- Quality Assurance

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)


### How Has This Been Tested?
Manually

### Screenshot
![](https://github.com/sachinsom93/temp/blob/master/bridge.jpg?raw=true)


### Checklist:
<!--**Delete irrelevant options.**-->

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials


**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

Rahulm2310 commented:
@sachinsom93 Can you please add a GIF showing the input flow instead of a static img.

Rahulm2310 commented:
@sachinsom93 Please fix it for login form as well.

sachinsom93 committed:
Type: BUG(ISSUE #202)
body:
    validated name, password, confirm password
    field in register page for spaces

codesankalp commented on src/register/Register.jsx:
```diff
@@ -156,6 +156,7 @@ export default function Register() {
                                     placeholder="Password"
                                     minLength={8}
                                     maxLength={64}
+                                    pattern="^[^\s].[a-zA-Z0-9/_]+$"
```
What about special characters in the password? It must not give validation when special characters like `@`is present in password.

mtreacy002 replied:
@Rahulm2310, @codesankalp and @sachinsom93 , here're my point of view on this:
1. the accepted regex pattern for validation needs to be consistent across backend and frontend sides of the application. In our case, since BIT is following MS backend rules on accepted regex for user's Registration, BIT backend and frontend must follow the same rules stated [here](https://github.com/anitab-org/mentorship-backend/blob/develop/app/utils/validation_utils.py).
2. @isabelcosta in [this zulip post](https://anitab-org.zulipchat.com/#narrow/stream/222534-mentorship-system/topic/Registration.20problem/near/189269896) had stated the reason behind why atm we don't use regex pattern for password (which is to make it easier for developer while we're still in the development stage - a different story when MS is in the production stage).  
3. MS android as MS frontend are also facing the same issue with Registration input validation which I raised [here](https://github.com/anitab-org/mentorship-android/issues/710). I also opened this Zulip [discussion](https://anitab-org.zulipchat.com/#narrow/stream/222534-mentorship-system/topic/Common.20Input.20validations/near/198847620) to start discussing potential solution. We still waiting for others' to give their feedback. 

Let me know what you think of this.

mtreacy002 replied:
I think we need to limit the scope of this pr to address only the bug stated in the issue #202 which is if the input consists of only spaces then show warning. We can use a pre-check to see if the input contains alphabets. We can leave the existing regex patterns untouched as  I believe already matched to the patterns used on MS backend.

mtreacy002 commented:
@sachinsom93 , can we please get an update on this?

sachinsom93 commented:
> @sachinsom93 , can we please get an update on this?

@mtreacy002 , working on the suggestions. Will update the pr soon :)

mtreacy002 commented:
@sachinsom93 , any update on this?

mtreacy002 commented:
Closing this issue as there is no activity.
